confluent_kafka/schema_registry/protobuf.py

```diff
@@ -0,0 +1,465 @@
+    |                                 |          | Registers schemas automatically if not               |
+    | auto.register.schemas           | bool     | previously associated with a particular subject.     |
+    |                                 |          | Defaults to True.                                    |
+    +---------------------------------|----------+------------------------------------------------------+
+    |                                 |          | Callable(SerializationContext, str) -> str           |
+    |                                 |          |                                                      |
+    | subject.name.strategy           | callable | Instructs the ProtobufSerializer on how to construct |
+    |                                 |          | Schema Registry subject names.                       |
+    |                                 |          | Defaults to topic_subject_name_strategy.             |
+    +---------------------------------+----------+------------------------------------------------------+
+    |                                 |          | Callable(SerializationContext, str) -> str           |
+    |                                 |          |                                                      |
+    | reference.subject.name.strategy | callable | Instructs the ProtobufSerializer on how to construct |
+    |                                 |          | Schema Registry subject names for Schema References  |
+    |                                 |          | Defaults to reference_subject_name_strategy          |
+    +---------------------------------+----------+------------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
+    +--------------------------------------+------------------------------+
+    | Subject Name Strategy                | Output Format                |
+    +======================================+==============================+
+    | topic_subject_name_strategy(default) | {topic name}-{message field} |
+    +--------------------------------------+------------------------------+
+    | topic_record_subject_name_strategy   | {topic name}-{record name}   |
+    +--------------------------------------+------------------------------+
+    | record_subject_name_strategy         | {record name}                |
+    +--------------------------------------+------------------------------+
+
+    See ``Subject name strategy`` for additional details.
+
+    Args:
+        message (Message): Protobuf Message.
```

Comment:
this reads like your passing an actual message to the constructor, not a type.

Reply:
renamed msg_type and used GeneratedProtocolMessageType as the type.